invenio_app_ils/ui/src/invenio_app_ils/common/api/loans/serializer.js

```diff
@@ -2,23 +2,30 @@ import _isEmpty from 'lodash/isEmpty';
 import { fromISO } from '../date';
 
 function serializeResponse(hit) {
-  const result = { ...hit.metadata };
+  let result = {};
   if (!_isEmpty(hit)) {
-    result['loan_pid'] = hit.id;
-    result['updated'] = fromISO(hit.updated);
+    result['id'] = hit.id;
     result['created'] = fromISO(hit.created);
+    result['updated'] = fromISO(hit.updated);
     result['availableActions'] = hit.links ? hit.links.actions : {};
-    const loan = hit.metadata;
-    if (!_isEmpty(loan)) {
-      result['item_pid'] = loan.item_pid ? loan.item_pid : '';
-      if (!_isEmpty(loan.item)) {
-        result['item'] = loan.item;
-      }
-      result['start_date'] = fromISO(loan.start_date);
-      result['end_date'] = fromISO(loan.start_date);
-      result['transaction_date'] = fromISO(loan.start_date);
+
+    if (hit.links) {
+      result['links'] = hit.links;
+    }
+    if (!_isEmpty(hit.metadata)) {
+      result['metadata'] = hit.metadata;
```

Comment:
While I go through the code again, I question myself if there is any actual reason that we flatten our responses, I understand the fact that we don't have to do `item.metadata.property` but by moving properties around we introduce yet another level of complexity instead of just accessing them where they are. @kprzerwa FDYWDS?

Reply:
that's why I don't flatten then anymore 👀